code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -536,3 +537,78 @@ get_true_breath_pressure(pp) --> gas_pp = pp/breath_pp*total_moles()
 
 		return TRUE
 	return FALSE
+
+
+/**
+ * Condenses the gases in the gas mixture into reagents in a areagent container
+ *
+ * Arguments:
+ * - [location][/atom]: The location the condensation is happening at
+ * - [reagents][/datum/reagents]: The reagent holder to output the condensed gases to
+ * - delta_time: The amount of time the condensation is happening over
+ */
+/datum/gas_mixture/proc/condense_gas_to(atom/location, datum/reagents/reagents, delta_time)
+	if(delta_time <= 0)
+		return
+
+	var/list/cached_gases = gases
+	if(!length(cached_gases))
+		return
+
+	var/cached_temp = return_temperature()
+	var/total_condensation_volume = 0
+	var/list/condensation_rates = list()
+	for(var/gas_id in cached_gases)
+		var/cond_max_rate = cached_gases[gas_id][GAS_META][META_GAS_COND_RATE]
+		if(!cond_max_rate)
+			continue
+
+		var/cond_max_temp = cached_gases[gas_id][GAS_META][META_GAS_COND_TEMP_MAX]
+		var/cond_min_temp = cached_gases[gas_id][GAS_META][META_GAS_COND_TEMP_MIN]
+		if(cached_temp >= cond_max_temp)
+			continue
+
+		var/cond_rate = clamp(LERP(0, cond_max_rate, (cond_max_temp - cached_temp) / (cond_max_temp - cond_min_temp)), 0, 1)
+		cond_rate = clamp(cached_gases[gas_id][MOLES] * DT_PROB_RATE(cond_rate, delta_time), 0, cached_gases[gas_id][MOLES])
```

Comment:
The condensation rate starts out as a ratio of the gas's moles that condense every second. This is calculated from the temperature of the gas and its minimum/maximum condensation temperatures. Lower is better and it caps out at the min... Actually I should really clamp that bit inside of the `LERP()` to prevent temperatures below `cond_min_temp` further increasing condensation rate. Anyway, this is then scaled using `DT_PROB_RATE()` by delta_time since it accurately reflects the scaling. This is then multiplied by the moles of the gas to get the final condensation rate.